lib/elixir_internal_certification/keyword/keywords.ex

```diff
@@ -25,10 +27,21 @@ defmodule ElixirInternalCertification.Keyword.Keywords do
   #     [%Keyword{}, ...]
 
   # """
-  def list_keywords(%User{} = user), do: Repo.all(KeywordQuery.list_keywords_by_user(user))
+  def list_keywords(%User{} = user) do
+    user
+    |> KeywordQuery.list_keywords_by_user()
+    |> Repo.all()
+    |> Repo.preload(:keyword_lookup)
+  end
 
   def get_keyword!(id), do: Repo.get!(Keyword, id)
 
+  def get_keyword!(%User{id: user_id} = _user, id) do
+    Keyword
+    |> Repo.get_by!(id: id, user_id: user_id)
+    |> Repo.preload(:keyword_lookup)
+  end
```

Comment:
```suggestion
  def get_keyword!(%User{id: user_id} = _user, id) do
    Keyword
    |> Repo.get_by(id: id, user_id: user_id)
    |> Repo.preload(:keyword_lookup)
  end
```
I think you may have to let this return `nil` and then you can handle that accordingly instead of returning an error. I think there will be cases where a user can type `keywords/<id>` into the url bar manually and it either doesn't exist or it's another user's keyword > both cases would error here. 

I guess returning to the main keywords page or a page saying it doesn't exist is better than an error and not handling it? 🤔